torchvision/models/vision_transformer.py

```diff
@@ -148,11 +160,31 @@ def __init__(
         self.representation_size = representation_size
         self.norm_layer = norm_layer
 
-        input_channels = 3
-
-        # The conv_proj is a more efficient version of reshaping, permuting
-        # and projecting the input
-        self.conv_proj = nn.Conv2d(input_channels, hidden_dim, kernel_size=patch_size, stride=patch_size)
+        if conv_stem_configs is not None:
+            # As per https://arxiv.org/abs/2106.14881
+            conv_proj = nn.Sequential()
+            prev_channels = 3
+            for i, conv_stem_layer_config in enumerate(conv_stem_configs):
+                conv_proj.add_module(
+                    f"conv_bn_relu_{i}",
+                    ConvNormActivation(
+                        in_channels=prev_channels,
+                        out_channels=conv_stem_layer_config.out_channels,
+                        kernel_size=conv_stem_layer_config.kernel_size,
+                        stride=conv_stem_layer_config.stride,
+                        padding=conv_stem_layer_config.padding,
+                        norm_layer=conv_stem_layer_config.norm_layer,
+                        activation_layer=conv_stem_layer_config.activation_layer,
+                        bias=conv_stem_layer_config.bias,
+                    ),
+                )
+                prev_channels = conv_stem_layer_config.out_channels
+            conv_proj.add_module(
+                "conv_last", nn.Conv2d(in_channels=prev_channels, out_channels=hidden_dim, kernel_size=1)
+            )
```

Comment:
[I'm writing the comment here in relation to the `nn.Sequential` [mypy](https://app.circleci.com/pipelines/github/pytorch/vision/14374/workflows/1f184498-f4f3-415d-96e0-d054cd565590/jobs/1156557) issue you face. Github doesn't let me reply to the original thread because the code is outdated.]

Try renaming `conv_proj` to `seq_proj` and after the final add_module, do:
`conv_proj: nn.Module = seq_proj`

The above should work but another alternative (which I'm not 100% will work) is doing after the final add_module a cast:
`conv_proj = typing.cast(nn.Module, conv_proj)`